### PR TITLE
WiP upgrade all dependencies; cleanup

### PR DESCRIPTION
**Pull request by SamuelMarks**

### CODE REVIEW

**Comment by AgustinCB** [comment]
Can you check the conflicts, please?

File: client/src/keys/mod.rs [hunk 1]
```diff
@@ -55,7 +56,9 @@ impl Password {
 pub fn generate_random_keypair() -> Result<(PublicKey, SecretKey), Error> {
     let context_flag = secp256k1::ContextFlag::Full;
     let context = secp256k1::Secp256k1::with_caps(context_flag);
-    let mut rng = OsRng::new().expect("OsRng");
+    let mut key = [0u8; 16];
+    OsRng.fill_bytes(&mut key);
+    let random_u64 = OsRng.next_u64();
     match context.generate_keypair(&mut rng) {
```
Comment (AgustinCB) [inline client/src/keys/mod.rs]: What happened with the `rng` variable?

Comment (SamuelMarks) [reply client/src/keys/mod.rs]: We should move to https://github.com/rust-random/getrandom

This branch is a WiP still

More info: https://rust-random.github.io/book/update-0.7.html

**Comment by SamuelMarks** [comment]
@AgustinCB Check https://github.com/Fantom-foundation/evm-rs/pull/4 as a replacement for this PR

**Pull request closed without merging**